### PR TITLE
Campaign creation form

### PR DESCRIPTION
**Pull request by caojohnny91**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,35 +1,45 @@
...
+import Navbar from "./components/Navbar/Navbar";
+import Dashboard from "./components/Dashboard/Dashboard";
+import Landing from "./components/Landing/Landing";
+import SignupForm from "./components/SignupForm/SignupForm";
+import SigninForm from "./components/SigninForm/SigninForm";
+import * as authService from "../src/services/authService";
+import CampaignForm from "./components/CampaignForm/CampaignForm";
 
 const App = () => {
   const [user, setUser] = useState(authService.getUser());
```
Comment (matthew-estes) [inline src/App.jsx:12]: need to initialize with an object

const [formData, setFormData] = useState({
    title: '',
    campaignType: '',
    description: '',
    goalAmount: '',
    endDate: ''
});

File: src/components/CampaignForm/CampaignForm.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      <main>
+        <h1>New Campaign Form</h1>
+        <form onSubmit={handleSubmit}>
+          <label htmlFor="title-input">Title </label>
+          <input
+            required
+            type="text"
+            name="title"
+            id="title-input"
+            onChange={handleChange}
```
Comment (matthew-estes) [inline src/components/CampaignForm/CampaignForm.jsx:27]: add value attribute to input:

 value={formData.title}

File: src/components/CampaignForm/CampaignForm.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          />
+
+          <label htmlFor="campaignType-input">Campaign Type </label>
+          <select
+            required
+            name="campaignType"
+            id="campaignType-input"
```
Comment (matthew-estes) [inline src/components/CampaignForm/CampaignForm.jsx:34]: add value attribute to input:

value={formData.campaignType}

File: src/components/CampaignForm/CampaignForm.jsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            <option value="Events">Events</option>
+            <option value="Environment">Environment</option>
+          </select>
+
+          <label htmlFor="description-input">Description </label>
+          <textarea
+            required
+            type="text"
+            name="description"
+            id="description-input"
```
Comment (matthew-estes) [inline src/components/CampaignForm/CampaignForm.jsx:52]: value={formData.description}

File: src/components/CampaignForm/CampaignForm.jsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            onChange={handleChange}
+          />
+
+          <label htmlFor="goalAmount-input">Goal Amount </label>
+          <input
+            required
+            type="Number"
+            name="goalAmount"
+            id="goalAmount-input"
```
Comment (matthew-estes) [inline src/components/CampaignForm/CampaignForm.jsx:61]: value={formData.goalAmount}

File: src/components/CampaignForm/CampaignForm.jsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            onChange={handleChange}
+          />
+
+          <label htmlFor="endDate-input">End Date </label>
+          <input
+            required
+            type="date"
+            name="endDate"
+            id="endDate-input"
```
Comment (matthew-estes) [inline src/components/CampaignForm/CampaignForm.jsx:70]: value={formData.endDate}

File: src/components/CampaignForm/CampaignForm.jsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            type="text"
+            name="description"
+            id="description-input"
+            onChange={handleChange}
+          />
+
+          <label htmlFor="goalAmount-input">Goal Amount </label>
+          <input
+            required
+            type="Number"
```
Comment (matthew-estes) [inline src/components/CampaignForm/CampaignForm.jsx:59]: type should be lowercase number

type="number"

**Review by matthew-estes — CHANGES_REQUESTED** [review]

**Review by craigbwagner — APPROVED** [review]

**Review by craigbwagner — APPROVED** [review]

**Pull request merged**